app/server/dao/src/main/java/io/syndesis/server/dao/manager/DataManager.java

```diff
@@ -442,4 +448,30 @@ public void clearCache() {
     private static <T> Function<ListResult<T>, ListResult<T>>[] noOperators() {
         return (Function<ListResult<T>, ListResult<T>>[]) NO_OPERATORS;
     }
+
+    private <T extends WithId<T>> void validateNoDuplicateName(final T entity) {
+        validateNoDuplicateName(entity, null);
+    }
+
+    private <T extends WithId<T>> void validateNoDuplicateName(final T entity, final String ignoreSelfId) {
+        if (WithName.class.isAssignableFrom(entity.getClass())) {
```

Comment:
why not 'entity instanceof WithName'?

Reply:
@chirino https://github.com/syndesisio/syndesis/pull/2173